dpdispatcher/ssh_context.py

```diff
@@ -110,19 +110,88 @@ def _setup_ssh(self):
         # machine = self.machine
         self.ssh = paramiko.SSHClient()
         self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy)
-        if self.totp_secret and self.password is None:
-            self.password = generate_totp(self.totp_secret)
-        self.ssh.connect(hostname=self.hostname, port=self.port,
-                        username=self.username, password=self.password,
-                        key_filename=self.key_filename, timeout=self.timeout,passphrase=self.passphrase,
-                        compress=True,
-                        )
-        assert(self.ssh.get_transport().is_active())
-        transport = self.ssh.get_transport()
-        transport.set_keepalive(60)
+        # if self.totp_secret and self.password is None:
+        #     self.password = generate_totp(self.totp_secret)
+        # self.ssh.connect(hostname=self.hostname, port=self.port,
+        #                 username=self.username, password=self.password,
+        #                 key_filename=self.key_filename, timeout=self.timeout,passphrase=self.passphrase,
+        #                 compress=True,
+        #                 )
+        # assert(self.ssh.get_transport().is_active())
+        # transport = self.ssh.get_transport()
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect((self.hostname, self.port))
+
+        #Make a Paramiko Transport object using the socket
+        ts = paramiko.Transport(sock)
+
+        #Tell Paramiko that the Transport is going to be used as a client
+        ts.start_client(timeout=10)
+
+        #Begin authentication; note that the username and callback are passed
+        if self.totp_secret:
+            ts.auth_interactive(self.username, self.inter_handler)
+        else:
+            default_path = os.path.join(os.environ["HOME"], ".ssh", "id_rsa")
+            path = ""
+            if self.key_filename:
+                path = os.path.abspath(self.key_filename)
+            if len(path) == 0:
+                path = default_path
+            try:
+                key = paramiko.RSAKey.from_private_key_file(path)
+            except paramiko.PasswordRequiredException:
+                key = paramiko.RSAKey.from_private_key_file(path, self.password)
+            try:
+                ts.auth_publickey(self.username, key)
+            except:
```

Comment:
Catch something. If everything is expected to be caught, use `except Exception`.